### PR TITLE
Issue #3064560 by Makishima: Makes the activity statistics block on the landing page extensible by other modules

### PR DESCRIPTION
**Pull request by VolodymyrMartsynenko**
## Problem
As a LU I want to see relevant statistics for Challenges and Ideas in the Activity overview block on the landing page

## Solution
Add the alter hook in the social_landing_page module to allow other modules to extend Activity overview block items

## Issue tracker
https://www.drupal.org/project/social/issues/3064560
https://jira.goalgorilla.com/browse/FOR-163

### CODE REVIEW

File: modules/social_features/social_landing_page/social_landing_page.api.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Provide a method to alter array of activity overview block items.
+ *
+ * @param array $build
+ *   Array with build data.
+ *
+ * @ingroup social_landing_page_api
+ */
+function hook_social_landing_page_activity_overview_alter(array &$build) {
```
Comment (OH17L) [inline modules/social_features/social_landing_page/social_landing_page.api.php]: Add some example code.

File: modules/social_features/social_landing_page/src/Plugin/Block/ActivityOverviewBlock.php [hunk 2]
```diff
@@ -274,6 +274,10 @@ public function build() {
       ],
     ];
 
+    \Drupal::moduleHandler()
```
Comment (OH17L) [inline modules/social_features/social_landing_page/src/Plugin/Block/ActivityOverviewBlock.php]: Use the injected service.

File: modules/social_features/social_landing_page/src/Plugin/Block/ActivityOverviewBlock.php [hunk 3]
```diff
@@ -274,6 +285,10 @@ public function build() {
       ],
     ];
 
+    $this->moduleHandler()
```
Comment (OH17L) [inline modules/social_features/social_landing_page/src/Plugin/Block/ActivityOverviewBlock.php]:
```suggestion
    $this->moduleHandler
```

**Comment by Kingdutch** [comment]
Hey I just saw this in the ECI demo and had some questions.

I think we add alter hooks a lot which opens up a large API surface that we have to support (essentially the entire block structure is now 'frozen'). Instead we should probably create a hook that simply requests additional stats in a specified format and just merges this. I'll add this as a discussion point to the Join OS Team Meeting.

Additionally we've implemented an alter hook that is at the end of the build output but this scenario would already be covered by Drupal's https://api.drupal.org/api/drupal/core%21modules%21block%21block.api.php/function/hook_block_view_alter/8.6.x which we've now duplicated.

**Comment by OH17L** [comment]
I think will be better to create new plugin type instead of a hook.

**Comment by OH17L** [comment]
Also, we can not use **hook_block_view_alter** because this block adding via block field widget which just calls **build** method.

**Comment by Kingdutch** [comment]
@OH17L 

Creating a statistic plugin (system) may actually not be a bad idea. It should allow us to control the caching, use it in different places and easily make it configurable what is displayed.

You could also use this for the other block types  and add a context (group/user) system.

The block instance configuration form could show the user all the plugins that support the current context. The plugin could then also have a `supportsChange` method that allows you to configure it to display the change of a statistic in a certain time period.

Is this something that is possibly already available in the Social KPI module? I think it could be a really cool feature that is useful in quite some places but it will require some architecture and might be outside of the scope for this story.

@jaapjan @ronaldtebrake thoughts?

**Comment by Kingdutch** [comment]
As for 
> Also, we can not use hook_block_view_alter because this block adding via block field widget which just calls build method.

This was fixed in https://www.drupal.org/project/block_field/issues/2978883 which is marked for review.

**Comment by ribel** [comment]
@VolodymyrMartsynenko could you create Issue on drupal.org, reference each other and update title here?

**Comment by jaapjan** [comment]
What about the feedback from @OH17L and @Kingdutch is this going to be processed? I do agree it would be nice, but I would be ok with a follow-up as well as that would require a lot more work to refactor.

**Comment by jaapjan** [comment]
Marking it as `needs work` since it would be better to implement the block_field patch instead of creating our own patch which does the same.

**Comment by ronaldtebrake** [comment]
The referenced PR: #1345 
Will actually already add this patch. 
I'll leave it up to you guys to either remove this issue as duplicate, or use this issue to create the statistic plugin system. Either way the patch should be able to solve the scope of this issue.
Let me know what you guys decide.

**Comment by ribel** [comment]
Since we now can use `hook_block_view_alter`, we don't need to have duplicated custom hook. So I think we can close this ticket. And we will refactor existing code in `social_challenge` module.
Also, I am leaving issue https://www.drupal.org/project/social/issues/3064560 with a solution to implement a plugin system when we will have time.

**Pull request closed without merging**